src/components/Sidebar/Sidebar.tsx

```diff
@@ -0,0 +1,85 @@
+import { Dispatch, SetStateAction, useEffect, useState } from "react";
+import ExploreSidebar from "./ExploreSidebar";
+import RequestSidebar from "./RequestSidebar";
+import { EnhancedPublicUser, PublicUser, User } from "../../utils/types";
+import mapboxgl from "mapbox-gl";
+import { viewRoute } from "../../utils/map/viewRoute";
+import { extend } from "lodash";
+import { HeaderOptions } from "../Header";
+import { trpc } from "../../utils/trpc";
+import _ from "lodash";
+import { Request } from "@prisma/client";
+
+interface SidebarProps {
+  sidebarType: HeaderOptions;
+  map: mapboxgl.Map;
+}
+
+const extendPublicUser = (
+  user: PublicUser,
+  favorites: PublicUser[],
+  received: Request[],
+  sent: Request[]
+): EnhancedPublicUser => {
+  return {
+    ...user,
+    isFavorited: favorites.some((favs) => favs.id === user.id),
+    incomingRequest: received.find((req) => req.fromUserId === user.id),
+    outgoingRequest: sent.find((req) => req.toUserId === user.id),
+  };
+};
+
+export const SidebarPage = (props: SidebarProps) => {
+  const { data: recommendations = [], refetch: refetchRecs } =
+    trpc.user.recommendations.me.useQuery();
+  const { data: favorites = [], refetch: refetchFavs } =
+    trpc.user.favorites.me.useQuery();
+  const {
+    data: requests = { sent: [], received: [] },
+    refetch: refetchRequests,
+  } = trpc.user.requests.me.useQuery();
+
+  useEffect(() => {
```

Comment:
from my understanding, since these refetches are not called anywhere other than on initial load, they aren't actually necessary. We can delete them, and TRPC will just fetch them when the component loads anyways. So we can remove this useEffect, and remove the refetches in the TRPC calls above.

Reply:
sounds good

Reply:
nvm, we do need those